jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -820,26 +115,56 @@ YUI.add('user-profile', function() {
             </div>
           </div>);
       }
+      */
+      var props = this.props;
       return (
         <div>
-          {this._generateRows('models', state.envList, state.loadingModels)}
-          {this._generateRows(
-            'bundles', state.bundleList, state.loadingBundles)}
-          {this._generateRows('charms', state.charmList, state.loadingCharms)}
-          <juju.components.AgreementList
-            getAgreements={this.props.getAgreements}
-            user={this.props.user} />
-          {this._generateRows(
-            'budgets', state.budgetList, state.loadingBudgets)}
+          <juju.components.UserProfileModelList
+            addNotification={props.addNotification}
+            canCreateNew={props.canCreateNew}
+            currentModel={props.currentModel}
+            env={props.env}
+            hideConnectingMask={props.hideConnectingMask}
+            jem={props.jem}
+            listModels={props.listModels}
+            showConnectingMask={props.showConnectingMask}
+            switchModel={props.switchModel}
+            user={props.user}
+            users={props.users} />
+          <juju.components.UserProfileEntityList
+            changeState={props.changeState}
+            charmstore={props.charmstore}
+            getDiagramURL={props.getDiagramURL}
+            type='bundle'
+            user={props.user}
+            users={props.users} />
+          <juju.components.UserProfileEntityList
+            changeState={props.changeState}
+            charmstore={props.charmstore}
+            getDiagramURL={props.getDiagramURL}
+            type='charm'
+            user={props.user}
+            users={props.users} />
+          <juju.components.UserProfileAgreementList
+            getAgreements={props.getAgreements}
+            user={props.user} />
+          <juju.components.UserProfileBudgetList
+            listBudgets={props.listBudgets}
+            user={props.user} />
         </div>);
     },
 
     render: function() {
       var username = this.props.user && this.props.user.usernameDisplay;
+      /* XXX Find some way to percolate these up from the child components. */
+      /*
```

Comment:
I'm guessing not. You and I had the discussion about how to support this API-wise, but I don't think there's a corresponding card. I'll be sure to create one.